pull/github.go

```diff
@@ -798,13 +798,47 @@ func (ghc *GitHubContext) loadPagedData() error {
 func (ghc *GitHubContext) loadCommits() ([]*Commit, error) {
 	log := zerolog.Ctx(ghc.ctx)
 
+	// GitHub's API is eventually consistent for commit pushed dates. We can
+	// sometimes get around this by using alternate APIs (see notes in
+	// loadCommitsOnce), but for some merge commits, that still fails. The only
+	// fix I know of in this case is to wait and try again.
+	//
+	// Retry with exponential backoff until it works or we hit the max total
+	// latency to avoid users thinking the bot got stuck or dropped an event.
+	const baseDelay = 1 * time.Second
+	const maxLatency = 20 * time.Second
```

Comment:
I think doing a time-bounded retry is a bit easier to think about than a retry count, but open to switching this to a more traditional N attempts where we manually compute the expected max latency.

Reply:
Is there a way we can add loggging to make it clear that our worker thread is in a long sleep? If for whatever reason we've failed a few times in a row, its hard to know why its not doing anything from the logs.